src/pages/workspace/reportFields/WorkspaceReportFieldsPage.tsx

```diff
@@ -0,0 +1,171 @@
+        }));
+    }, [filteredPolicyFieldList, policy, selectedReportFields]);
+
+    const updateSelectedReportFields = (item: ReportFieldForList) => {
+        const fieldKey = ReportUtils.getReportFieldKey(item.fieldID);
+        const updatedReportFields = selectedReportFields.find((selectedReportField) => selectedReportField.name === item.value)
+            ? selectedReportFields.filter((selectedReportField) => selectedReportField.name !== item.value)
+            : [...selectedReportFields, filteredPolicyFieldList[fieldKey]];
+        setSelectedReportFields(updatedReportFields);
+    };
+
+    const isLoading = reportFieldsList === undefined;
+    const shouldShowEmptyState = Object.values(filteredPolicyFieldList).length <= 0 && !isLoading;
+
+    const getHeaderButtons = () => (
+        <View style={[styles.w100, styles.flexRow, styles.gap2, isSmallScreenWidth && styles.mb3]}>
+            <Button
+                medium
+                success
+                onPress={() => {}}
+                icon={Expensicons.Plus}
+                text={translate('workspace.reportFields.addField')}
+                style={[isSmallScreenWidth && styles.flex1]}
+            />
+        </View>
+    );
+
+    const getCustomListHeader = () => (
+        <View style={[styles.flex1, styles.flexRow, styles.justifyContentBetween, styles.pl3, styles.pr9]}>
+            <Text style={styles.searchInputStyle}>{translate('common.name')}</Text>
+            <Text style={[styles.searchInputStyle, styles.textAlignCenter]}>{translate('common.type')}</Text>
+        </View>
+    );
+
+    const getHeaderText = () => (
+        <View style={[styles.ph5, styles.pb5, styles.pt3]}>
+            <Text style={[styles.textNormal, styles.colorMuted]}>{translate('workspace.reportFields.subtitle')}</Text>
+        </View>
+    );
+
+    return (
+        <AccessOrNotFoundWrapper
+            policyID={policyID}
+            accessVariants={[CONST.POLICY.ACCESS_VARIANTS.ADMIN, CONST.POLICY.ACCESS_VARIANTS.PAID]}
+            featureName={CONST.POLICY.MORE_FEATURES.ARE_REPORT_FIELDS_ENABLED}
+        >
+            <ScreenWrapper
+                includeSafeAreaPaddingBottom={false}
+                style={[styles.defaultModalContainer]}
+                testID={WorkspaceReportFieldsPage.displayName}
+                shouldShowOfflineIndicatorInWideScreen
+                offlineIndicatorStyle={styles.mtAuto}
+            >
+                <HeaderWithBackButton
+                    icon={Illustrations.Pencil}
+                    title={translate('workspace.common.reportFields')}
+                    shouldShowBackButton={isSmallScreenWidth}
+                >
+                    {!isSmallScreenWidth && getHeaderButtons()}
+                </HeaderWithBackButton>
+                {isSmallScreenWidth && <View style={[styles.pl5, styles.pr5]}>{getHeaderButtons()}</View>}
```

Comment:
Why we didn't use the `WorkspacePageWithSections` component? It is used in almost every workspace-related page.